### ISSUE TITLE
code for copulas --- suggestion

### ISSUE THREAD

**Issue by jstac** [QuantEcon/QuantEcon.py#158]
There's limited code in Python for working with or simulating copulas.  This library

http://firsttimeprogrammer.blogspot.com.au/2015/02/copulalib-how-to-use-copulas-in-python.html
https://pypi.python.org/pypi/copulalib/1.1.0

has some functionality but is tied to outdated dependencies.  That code could be used as a starting point in QuantEcon with dependencies only in more modern packages.

**Comment by jstac** [comment]
It would be useful if the copulas could be initialized directly with parameters, rather than forcing the user to initialize them with data (from which parameters are estimated).

**Comment by mmcky** [comment]
@jstac I would be keen to help with this package. I will need to do a bit of homework to get up to speed. There is a recent paper motivating my interest in this topic. 

Chessa A, Crimaldi I, Riccaboni M, Trapin L (2014) Cluster Analysis of Weighted Bipartite Networks: A New Copula-Based Approach. PLoS ONE 9(10): e109507. doi:10.1371/journal.pone.0109507

The ``copulalib`` code seems to have been merged into the ``ambhas`` package (https://pypi.python.org/pypi/ambhas/).

**Comment by jstac** [comment]
@mmcky Right.  `ambhas` also has old dependencies and a bunch of stuff we wouldn't need so it would still be valuable to put together our own code.  There's a fair bit of R code for copulas lying around, as well as the code in ambhas.  The R code could be translated pretty easily.

**Comment by spencerlyon2** [comment]
Before looking too much at the R code we should make sure it is non GPL, a lot of R code is...

**Comment by Guo-Zhang** [comment]
@sglyon It is GPL>=3. My team will work with this issue these days.
